### PR TITLE
Fixed #28863 -- Fixed filter on annotation that contains Q.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/28863

### CODE REVIEW

File: django/db/models/sql/where.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def as_sql(self, compiler, connection):
                 # around the inner SQL in the negated case, even if the
                 # inner SQL contains just a single expression.
                 sql_string = 'NOT (%s)' % sql_string
-            elif len(result) > 1:
+            else:
```
Comment (adamchainz) [inline django/db/models/sql/where.py]: all the test failures seem related to this

**Comment by adamchainz** [comment]
LGTM apart from SQL changes. But I can see why the parens might always be needed in order to ensure the SQL always compiles.

File: django/db/models/sql/where.py (hunk 1 above)
Comment (sir-sigurd) [reply django/db/models/sql/where.py]: Without it there are failures on Postgres.

Comment (adamchainz) [reply django/db/models/sql/where.py]: the failures now are just assertions on generated SQL so pretty simple fixes

**Comment by sir-sigurd** [comment]
buildbot, test on oracle.

**Comment by timgraham** [comment]
The error for the new test on Oracle `django.db.utils.DatabaseError: ORA-00923: FROM keyword not found where expected`.

**Comment by felixxm** [comment]
Using `boolean` type in `SELECT` is not supported on Oracle. The only way to emulate this is to use `CASE ... WHEN ..` syntax:
```sql
SELECT COUNT(*)
FROM (
    SELECT
        "EXPRESSIONS_COMPANY"."ID" AS Col1,
        (CASE WHEN "EXPRESSIONS_COMPANY"."NUM_EMPLOYEES" > :arg1 THEN 1 ELSE 0 END) AS "NUM_EMPLOYEES_CHECK"
    FROM "EXPRESSIONS_COMPANY"
    WHERE
        (CASE WHEN "EXPRESSIONS_COMPANY"."NUM_EMPLOYEES" > :arg1 THEN 1 ELSE 0 END) > :arg1) = :arg0
    GROUP BY
        "EXPRESSIONS_COMPANY"."ID",
        (CASE WHEN "EXPRESSIONS_COMPANY"."NUM_EMPLOYEES" > :arg1 THEN 1 ELSE 0 END)
    ) subquery
```

**Comment by timgraham** [comment]
If besides this issue, the 2.0 release is ready later today, I don't think I'll block the release on it. It can always be fixed in a later bug fix release.

**Comment by felixxm** [comment]
We can skip this test on Oracle, I'm fine with that.

**Comment by adamchainz** [comment]
Thanks @sir-sigurd @tim-schilling @felixxm :)

**Pull request merged**